jpegenc/models/rgb2ycbcr/rgb2ycbcr.py

```diff
@@ -0,0 +1,156 @@
+#!/bin/python
+from myhdl import block, Signal, ResetSignal, intbv, delay, instance, always_comb, always_seq
+from myhdl.conversion import analyze
+
+Y_COEFF=[0.2999,0.587,0.114]
+CB_COEFF=[-0.1687,-0.3313,0.5]
+CR_COEFF=[0.5,-0.4187,-0.0813]
+OFFSET=[0,128,128]
+
+def build_coeffs(fract_bits):
+
+    Y=[int(round(Y_COEFF[i]*(2**fract_bits )))for i in range(3)]
+    Cb=[int(round(CB_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+    Cr=[int(round(CR_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+    Offset=[int(round(OFFSET[i]*(2**fract_bits ))) for i in range(3)]
+
+    return Y,Cb,Cr,Offset
+
+
+class RGB(object):
+
+    def __init__(self, nbits=8):
+        self.nbits=nbits
+        self.red = Signal(intbv(0)[nbits:])
+        self.green = Signal(intbv(0)[nbits:])
+        self.blue = Signal(intbv(0)[nbits:])
+
+    def bitLength(self): return self.nbits
+
+
+class YCbCr(object):
+
+    def __init__(self, nbits=8):
+        self.nbits = nbits
+        self.y = Signal(intbv(0)[nbits:])
+        self.cb = Signal(intbv(0)[nbits:])
+        self.cr = Signal(intbv(0)[nbits:])
+
+    def bitLength(self): return self.nbits
+
+@block
+def rgb2ycbcr(ycbcr, enable_out, rgb, enable_in, clk, reset, fract_bits = 14, nbits = 8):
```

Comment:
Because there are a few of us working on this we need to all be consistent on `clock`, `reset` and port ordering.  The use of interfaces is going to help considerably with readability, testing, and design.

Is `enable_in` and `enable_out` used a *data is good* / *data valid*?  If so we should build them into the interface.

Then the port would look like:
```python
def rgb2ycbcr(
    # ports (inputs interface, output interfaces, ports)
    rgb, ycbcr, clock, reset
    # parameters 
    # ...
)
```

Note, you don't need to inline the comments, I did that for additional explanation you can keep it as

```python
rgb2ycbcb(rgb, ycbcr, clock, reset, ...)
```

For the parameters `nbits` isn't needed, you can get that from the inputs, we should turn `frac_bits` from a hardware parameter to a domain parameter, when we specify 14 `frac_bits` what are we specifying (I don't mean we are defining 14 fraction bits).  Are we defining the color depth in the new color mode?